Dockerfile

```diff
@@ -1,85 +1,23 @@
 FROM debian:buster-slim
 
-# Toolchain dependencies for Unvanquished, Daemon, external_deps, or build-release
-ENV TOOLCHAIN_DEPS=' \
-    autoconf \
-    cmake \
-    curl \
-    g++ \
-    g++-mingw-w64-i686 \
-    g++-mingw-w64-x86-64 \
-    git \
-    libtool \
-    make \
-    p7zip-full \
-    python2 \
-    python3-jinja2 \
-    python3-yaml \
-    rsync \
-'
-
-# Libraries statically linked into the Linux binaries of Daemon
-ENV DAEMON_LINUX_STATIC_DEPS=' \
-    libfreetype6-dev \
-    libgmp-dev \
-    libjpeg-dev \
-    libogg-dev \
-    libopus-dev \
-    libopusfile-dev \
-    libpng-dev \
-    libvorbis-dev \
-    libwebp-dev \
-    nettle-dev \
-    zlib1g-dev \
-'
-
-# These are needed for building Daemon or its dependencies but not statically linked into the binary.
-# The display-related ones are relevant for SDL configuration and are taken from
-# https://wiki.libsdl.org/FAQLinux#How_do_I_get_all_the_dependencies_for_building_SDL_on_Ubuntu.3F
-# The audio ones are relevant for OpenAL configuration.
-ENV DAEMON_LINUX_AV_DEPS=' \
-    libasound2-dev \
-    libaudio-dev \
-    libgl1-mesa-dev \
-    libjack-jackd2-dev \
-    libpulse-dev \
-    libx11-dev \
-    libxcursor-dev \
-    libxext-dev \
-    libxi-dev \
-    libxinerama-dev \
-    libxrandr-dev \
-    libxss-dev \
-    libxxf86vm-dev \
-'
+ARG rev
```

Comment:
I think for better caching this should be after the apt getting (mostly because the Unvanquished source is a smaller download). And in any case it should be right before the git clone step.

Reply:
I split the helper in various script in `docker-helpers/` folder, and reordered things in the Docker file.